jaxopt/_src/prox.py

```diff
@@ -70,8 +70,12 @@ def prox_lasso(x: Any,
   if l1reg is None:
     l1reg = 1.0
 
-  fun = lambda u, v: jnp.sign(u) * jax.nn.relu(jnp.abs(u) - v * scaling)
-  return tree_util.tree_map(fun, x, l1reg)
+  if type(l1reg) == float:
+    fun = lambda u: jnp.sign(u) * jax.nn.relu(jnp.abs(u) - l1reg * scaling)
+    return tree_util.tree_map(fun, x)
```

Comment:
There is a lot of code duplication. I think it would be more elegant to do
```python
if type(l1reg) == float:
  l1reg = tree_util.tree_ones_like(x) * l1reg
```

Reply:
I simplified it.